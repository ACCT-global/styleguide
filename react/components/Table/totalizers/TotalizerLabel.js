import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

class TotalizerLabel extends PureComponent {
  render() {
    const { label } = this.props

    return <div className="c-muted-2 f6 mb2">{label}</div>
  }
}

TotalizerLabel.propTypes = {
  label: PropTypes.string.isRequired,
}

export default TotalizerLabel
