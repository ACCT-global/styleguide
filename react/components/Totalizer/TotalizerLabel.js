import React from 'react'
import PropTypes from 'prop-types'

const TotalizerLabel = ({ label, mobileScroll, testId }) => (
  <div
    data-testid={testId}
    className={`c-muted-2 f6 ${mobileScroll ? 'nowrap' : ''}`}
  >
    {label}
  </div>
)

TotalizerLabel.propTypes = {
  label: PropTypes.string.isRequired,
  mobileScroll: PropTypes.bool,
  testId: PropTypes.string,
}

export default TotalizerLabel
