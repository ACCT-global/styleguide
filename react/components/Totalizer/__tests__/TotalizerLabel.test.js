import React from 'react'
import { render } from 'react-testing-library'

import TotalizerLabel from '../TotalizerLabel'

describe('Totalizer/Label spec', () => {
  const testId = 'testing_id'
  const label = 'testing_label'
  const props = {
    label,
    testId,
  }

  it('should match snapshot on desktop', () => {
    const Label = render(<TotalizerLabel {...props} />)
    expect(Label.asFragment()).toMatchSnapshot()
  })

  it('should match snapshot on mobile', () => {
    const Label = render(<TotalizerLabel {...props} mobileScroll />)
    expect(Label.asFragment()).toMatchSnapshot()
  })

  it('should render correctly', () => {
    const Label = render(<TotalizerLabel {...props} />)
    const ElementToTest = Label.getByTestId(testId)
    expect(ElementToTest).toBeTruthy()
    expect(ElementToTest.childElementCount).toBe(0)
    expect(ElementToTest.textContent).toBe(label)
  })
})
