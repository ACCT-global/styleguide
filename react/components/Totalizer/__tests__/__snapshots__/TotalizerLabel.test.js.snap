// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Totalizer/Label spec should match snapshot on desktop 1`] = `
<DocumentFragment>
  <div
    class="c-muted-2 f6 "
    data-testid="testing_id"
  >
    testing_label
  </div>
</DocumentFragment>
`;

exports[`Totalizer/Label spec should match snapshot on mobile 1`] = `
<DocumentFragment>
  <div
    class="c-muted-2 f6 nowrap"
    data-testid="testing_id"
  >
    testing_label
  </div>
</DocumentFragment>
`;
